ImportXml.py

```diff
@@ -0,0 +1,182 @@
+# flake8: noqa
+import re
+import sys
+from json import dump
+
+from bs4 import BeautifulSoup
+
+# import pandas as pd
+# from collections import defaultdict
+
+
+def read_questionnaire_structure(filepath: str, writefiles: bool):
+    """
+    Reads questionnaire structure XML file
+
+    Args:
+        filepath: A string with a path to a XML file with questionnaire structure
+        writefiles: if True will dumps data to files as .json which can be opened \
+                    by json.load(file) or pandas.read_json(file).
+    Returns:
+        A dict containing the questionnaire structure:
+        {
+            "sections": {
+               {
+                   "id":13901,
+                   "title": "Demographics",
+                   "info": "In this section, we will ask general questions about yourself \
+                       and your doctoral project."
+               },
+               ....
+            },
+            "questions": {
+                "name": "A1",
+                "question": "What is your year of birth?",
+                "description": "Please use the following format: 'xxxx' , e.g. 1989",
+                "type": "category",
+                "categories": {"I don't want to answer this question": 'A2',
+                                ...
+                                'Year of birth: ': 'A4other'},
+                "subquestions": {
+                    'E7a_SQ001': 'My supervisor is well informed about my field of research.',
+                    'E7a_SQ002': 'My supervisor is available when I need advice.',
+                    'E7a_SQ003': 'My supervisor is open to and respects my research ideas.',
+                    'E7a_SQ004': 'My supervisor gives constructive feedback.',
+                    ...},
+                "section_id": 13901
+            }
+        }
+
+    Raises:
+        ValueError: Invalid XML file
+        ValueError: Provided file has unexpected structure
+    """
+
+    def uniCodeRe(s) -> str:
+        # '\u00a0','\u20ac', '\u00b2', '\u00a7'
+        matched = ["\n", "\t", "\xa0"]
+        # if you want to change the unicode
+        # s_new = s.replace(u'\u2018', '"').replace(u'\u2019', '\'').replace(u'\u00f6', r'ö').replace(u'\u00fc', r'ü')
+        # s_new = s
+        res_s = re.sub("|".join(matched), "", s)
+        return res_s
+
+    def textParse(bsString) -> str:
+        text = ""
+        sps = BeautifulSoup(bsString, "html.parser").find_all("p")
+        # sbs = BeautifulSoup(bsString, 'html.parser').find_all('b')
+        for sp in sps:
+            if bool(sp.string):
+                text += uniCodeRe(sp.string)
+        return text
```

Comment:
ye, good idea, I think I should integrate all the html parse function as one, but I think cannot directly use this function. Because in the file, the text structures under html tag are different from one part to another part.
Anyway, I will try to merge the html parsing part into one function.